config/v1alpha1/types_image_policy.go

```diff
@@ -143,6 +151,43 @@ type PolicyFulcioSubject struct {
 	SignedEmail string `json:"signedEmail"`
 }
 
+// PKI defines the root of trust based on Root CA(s) and corresponding intermediate certificates.
+type PKI struct {
+	// caRootsData contains base64-encoded data of a certificate bundle PEM file, which contains one or more CA roots in the PEM format. The total length of the data must not exceed 8192 bytes.
+	// +required
+	// +kubebuilder:validation:MaxLength=8192
+	// +kubebuilder:validation:XValidation:rule="string(self).startsWith('-----BEGIN CERTIFICATE-----')",message="the caRootsData must start with base64 encoding of '-----BEGIN CERTIFICATE-----'."
+	// +kubebuilder:validation:XValidation:rule="string(self).endsWith('-----END CERTIFICATE-----\\n') || string(self).endsWith('-----END CERTIFICATE-----')",message="the caRootsData must end with base64 encoding of '-----END CERTIFICATE-----'."
+	CertificateAuthorityRootsData []byte `json:"caRootsData"`
+	// caIntermediatesData contains base64-encoded data of a certificate bundle PEM file, which contains one or more intermediate certificates in the PEM format. The total length of the data must not exceed 8192 bytes.
+	// caIntermediatesData requires CertificateAuthorityRoots to be set.
```

Comment:
Validate this with CEL

```
has(self.caIntermediatesData) ? has(self.caRootsData) : true
```

Also, you are using the go type name for `CertificateAuthorityRoots`, use the serialised version

Reply:
I tested it a bit, maybe it's not necessary? it seems we can not trigger validation if required caRootsData hasn't being set
```
spec:
  scopes:
  - example.com
  policy:
    rootOfTrust:
      policyType: PKI
      pki:
        caRootsData: ****
        pkiCertificateSubject:
          email: test-user@example.com
          hostname: invalid-email

The ClusterImagePolicy "pki-policy" is invalid: 
* spec.policy.rootOfTrust.pki.caRootsData: Required value
* <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation
```

Reply:
You're right, caRootsData is required so this isn't needed, good point